src/main/java/ch/bfh/bti7081/s2018/green/views/MedicationPrescriptionView.java

```diff
@@ -2,89 +2,145 @@
             panel = new Panel("New Medication");
+            med = new Medication();
         } else {
-        	panel = new Panel("Edit Medication");
+            panel = new Panel("Edit Medication");
         }
         this.medication = med;
-        bindMedication(med);
+        bindMedication(this.medication);
         this.setModal(true);
+        //Panel panel = new Panel("Medication prescription");
         CustomLayout panelContent = new CustomLayout("medicationPrescription");
         panelContent.addComponent(medName, "medName");
         panelContent.addComponent(medStartDate, "medStartDate");
         panelContent.addComponent(medStopDate, "medStopDate");
-
         panelContent.addComponent(medPeriod, "medFrequency");
         panelContent.addComponent(medDose, "medDose");
         panelContent.addComponent(medPrescriberFullName, "medPrescriber");
         panelContent.addComponent(btnSave, "medSaveButton");
         panel.setContent(panelContent);
-        setContent(panel);
-        new MedicationPrescriptionPresenter(this);
+        this.setContent(panel);
+
+        // make sure an element can just be edited 20 minutes long after start time of medication
+        if (isEditMode && !this.medication.getStartDate().isAfter(LocalDateTime.now().minusMinutes(20))) {
+            medName.setEnabled(false);
+            medStartDate.setEnabled(false);
+            medPeriod.setEnabled(false);
+            medDose.setEnabled(false);
+        }
+        new MedicationPrescriptionPresenter(this, viewBehind);
     }
-    
-    private void bindMedication(Medication medication) {
-        binder = new Binder<>();
-        binder.forField(medName).bind(Medication::getName, Medication::setName);
-        binder.forField(medStartDate).bind(Medication::getStartDate, Medication::setStartDate);
-        binder.forField(medStopDate).bind(Medication::getEndDate, Medication::setEndDate);
-        // TODO: support integers: https://vaadin.com/docs/v8/framework/datamodel/datamodel-forms.html#datamodel.forms.conversion
-        //binder.forField(medPeriod).bind(Medication::getPeriode, Medication::setPeriode);
-        //binder.forField(medDose).bind(Medication::getDose, Medication::setDose);
-        //binder.forField(medPrescriberFullName).bind(Medication::getFullName, Medication::setEndDate);
 
+    // performs field validation for new records
+    private void bindMedication(Medication medication) {
+        this.validateFields();
         binder.readBean(medication);
     }
 
+    //perform field validation when saving a record - for values that could only be validated when editing an existing record
     public Medication getMedication() {
-        // TODO: validate input
         try {
-            binder.writeBean(this.medication);
-            System.out.println(medication.getStartDate());
-            System.out.println(LocalDateTime.now());
-            
+            this.validateFields();
+            if (medStartDate.getValue() != null)
```

Comment:
When can this be null? Shouldn't there be an `asRequired` on the field?

Reply:
Fixed